client/Android/FreeRDPCore/jni/android_freerdp.c

```diff
@@ -939,19 +1030,28 @@ JNIEXPORT void JNICALL jni_freerdp_set_sound_redirection(JNIEnv *env,
 			redirect ? ((redirect == 1) ? "Server" : "Redirect") : "None");
 
 	settings->AudioPlayback = (redirect == 2) ? TRUE : FALSE;
-	settings->RemoteConsoleAudio = (redirect == 1) ? TRUE : FALSE;
 	if (settings->AudioPlayback)
 	{
+		int ret;
 		p = malloc(sizeof(char*));
+		if (!p)
+		{
+			settings->AudioPlayback = FALSE;
+			return JNI_FALSE;
+		}
 		p[0] = "rdpsnd";
```

Comment:
Not your code but it would be much nicer to just declare+assign p like this:
```
char* p[1] = { "rdpsnd" };
```
No need to malloc/check p